Extend `wrap` functionality to non-function components

AbigailMcP opened the pull request:
This addresses issue #101. Please see comments on the thread for a more in depth discussion of the changes being made.

**Currently** 
For elements like `Text`,  the React Native renderer returns an instance of a `ReactNativeFiberHostComponent`. This is what is passed to the ref callback; none of the original props are exposed.

Therefore, if you pass a Cavy ref to a `Text` component, anything other than the presence of that element on the page is impossible to test. You cannot call any props after finding the component using `findComponent`.

**Proposed**
You can pass a `Text` component into wrap and Cavy will return a wrapped version of that component with accessible props.

If you attempt to wrap a proper class component, Cavy warns you that you do not need to do this and just returns the component.

**Thoughts**
I've also added `containsText` as a helper function to the `TestScope`, but all this does is check for `component.props.children.includes(text)`. @jalada I could change the name of this to be more generic?

AbigailMcP commented:
@jalada - when you're happy with this I can make the necessary changes to the Cavy documentation.

jalada commented on src/wrap.js:
```diff
@@ -11,31 +14,85 @@ import { useImperativeHandle, forwardRef } from 'react';
+  }
+
+  if (typeof Component == 'object') {
+    class WrapperComponent extends React.Component {
```
Is this really the minimum we have to do to wrap it? We don't have to hoist statics or anything weird like that?

(sorry if this is a silly question)

AbigailMcP replied:
> Happy with the name `containsText` as it's nice and friendly.

👌 

> API all looks fine, so feel free to start updating the documentation too.

Will do!

> Dang, we could really do with #147. I got stuck in that PR getting the build working, I wonder if it would be easier now we've got the sample app more up-to-date? 🤷

Yeah I thought this too as I wanted to test passing in class components into `wrap` i.e. incorrect use-cases. I'll try building it today, but I'd like to get this in regardless.

> I can see a test for `wrap(Text)`, but do we test that `wrap(FunctionComponent)` works anywhere in the sample app? [This ActionButton test?](https://github.com/pixielabs/cavy/blob/extend-wrap/sample-app/CavyDirectory/app/ActionBar.js#L33) If so, I'm happy for this to go in.

We test that the button is there `await spec.exists('ActionBar.EmailButton')` - I'll add one that interacts with the button's props.

> Have you thought about the version bump? I don't think we'll break anyone's code with this unless they were doing some funky workaround, so... minor bump?

I was thinking minor bump. If people were working around this, then their workarounds should still be valid.

AbigailMcP replied:
Have hoisted statics.

jalada commented on src/wrap.js:
```diff
@@ -11,31 +14,85 @@ import { useImperativeHandle, forwardRef } from 'react';
+    return WrapperComponent;
+  }
+
+  const message = "Looks like you're passing a class component into `wrap` - " +
```
Are we able to log the name of the component?

AbigailMcP replied:
Should be able to!

AbigailMcP replied:
Ok so this is difficult because component here is a reference to a factory function.

jalada replied:
OK no worries

AbigailMcP commented:
> Happy with the name `containsText` as it's nice and friendly.

👌 

> API all looks fine, so feel free to start updating the documentation too.

Will do!

> Dang, we could really do with #147. I got stuck in that PR getting the build working, I wonder if it would be easier now we've got the sample app more up-to-date? 🤷

Yeah I thought this too as I wanted to test passing in class components into `wrap` i.e. incorrect use-cases. I'll try building it today, but I'd like to get this in regardless.

> I can see a test for `wrap(Text)`, but do we test that `wrap(FunctionComponent)` works anywhere in the sample app? [This ActionButton test?](https://github.com/pixielabs/cavy/blob/extend-wrap/sample-app/CavyDirectory/app/ActionBar.js#L33) If so, I'm happy for this to go in.

We test that the button is there `await spec.exists('ActionBar.EmailButton')` - I'll add one that interacts with the button's props.

> Have you thought about the version bump? I don't think we'll break anyone's code with this unless they were doing some funky workaround, so... minor bump?

I was thinking minor bump. If people were working around this, then their workarounds should still be valid.

AbigailMcP commented on src/wrap.js:
```diff
@@ -11,31 +14,85 @@ import { useImperativeHandle, forwardRef } from 'react';
+    });
+  }
+
+  if (typeof Component == 'object') {
```
👌

AbigailMcP commented:
@jalada - addressed the issues above except for the component name logging problem. I've added an extra test for accessing the props of the wrapped function component `ActionButton`.

jalada commented:
Great! This is looking good to go :)
